test/integration/tests/build-ghc/Main.hs

```diff
@@ -0,0 +1,29 @@
+import StackTest
+import System.Directory (withCurrentDirectory)
+
+main :: IO ()
+main = do
+    stack ["new", "--resolver=lts-13.11", "tmpPackage"]
+
+    -- use a commit which is known to succeed with hadrian binary-dist
+    let commitId = "cd85f8a71bb56cff332560e1d571b3406789fb71"
+        flavour  = "quick"
+
+    withCurrentDirectory "tmpPackage" $ do
+       appendFile "stack.yaml" $ unlines
+         [ "compiler-build:"
+         , "  git: https://gitlab.haskell.org/ghc/ghc.git"
+         , "  commit: " ++ commitId
+         , "  flavour: " ++ flavour
+         , ""
+         , "compiler: ghc-git-" ++ commitId ++ "-" ++ flavour
+         ]
+
+       -- Setup the package
+       stack ["setup"]
```

Comment:
I wonder how long does it take to run this test? Probably a couple of hours? Not sure if it will be OK to run on every run of integration tests...

Reply:
Indeed. I am not sure we should enable this test by default. A GHC build with the "quick" flavour takes ~28min on my Intel 9700k (it took ~2h39 with my previous Intel i7-970).

----

For the record: a lot of time is wasted generating the documentation, so I've proposed a patch (https://gitlab.haskell.org/ghc/ghc/merge_requests/354) that:
1) allows flavours to disable doc generation
2) uses this feature to disable doc generation with the "quickest" flavour

As you can see in the discussion, it wasn't welcome. But a few days after point (1) has been reimplemented by someone else and merged without any discussion (https://gitlab.haskell.org/ghc/ghc/merge_requests/461)...

So now we could define and use a faster flavour for this test. We can host a GHC repository with the additional faster flavour somewhere or alternatively we could try to submit a GHC MR to implement point (2) now that the counter arguments are moot (or alternatively we could propose a new "quickestest" flavour).

(Just as I'm writing this, I've found a bug in Hadrian: when we disable doc generation the binary-dist target fails...)

----

Note that currently Hadrian generates invalid bindists so I'm waiting for https://gitlab.haskell.org/ghc/ghc/merge_requests/651 to be merged to get the test passing... Currently without this patch the installed GHC crashes.

Reply:
I'll add a feature to `master` to have a "superslow" flag to the integration test suites, and this test can be part of that. Sound reasonable?

Reply:
Feature added in 4b9f1b14a87f210fd87fc40dbaf1ea108a68aa43. If you merge/rebase with master, you can add `superslow $` in front of the test code.

Reply:
Done. I have modified it a little to make ``run-single-test.sh`` work (it doesn't specify a speed)